src/player.js

```diff
@@ -2132,18 +2132,25 @@ export default class Player extends FakeEventTarget {
    */
   _markActiveTrack(track: Track): void {
     let tracks;
+    let markedActive = false;
     if (track instanceof VideoTrack) {
       tracks = this._getVideoTracks();
     } else if (track instanceof AudioTrack) {
       tracks = this._getAudioTracks();
     } else if (track instanceof TextTrack) {
       tracks = this._getTextTracks();
     }
+    // add label check for native player that doesn't parse the video tracks from beginning.
     if (tracks) {
       for (let i = 0; i < tracks.length; i++) {
-        tracks[i].active = track.index === tracks[i].index;
+        tracks[i].active = track.index >= 0 ? track.index === tracks[i].index : track.label === tracks[i].label;
+        markedActive = markedActive ? markedActive : tracks[i].active;
       }
     }
+    // selected track for some cases won't be available on the parsed tracks, added to get them from the API.
+    if (!markedActive) {
+      this._tracks.push(track);
+    }
```

Comment:
this is not really needed as in native there is always one active track so this API doesn't even really make sense.
Unless I'm missing something - this is not required and the incentive for adding the track change event was only for analytics purposes where rendition changes are what maters.
We can then see for example correlation between buffering events and track changes or see excessive rendition changes which can show some bad encoding ladder config.

Reply:
Native video tracks have one video track on the beginning after parsing, so on the change track, we don't have the track after parsing so we need to add the selected tracks to `_tracks` object for other API's. (getActiveTrack, getVideoTracks)